crates/chia-secp/src/secp256k1/public_key.rs

```diff
@@ -0,0 +1,59 @@
+use std::fmt;
+use std::hash::{Hash, Hasher};
+
+use chia_sha2::Sha256;
+use k256::ecdsa::signature::hazmat::PrehashVerifier;
+use k256::ecdsa::{Error, VerifyingKey};
+
+use super::Secp256k1Signature;
+
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct Secp256k1PublicKey(pub(crate) VerifyingKey);
+
+impl Hash for Secp256k1PublicKey {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.to_bytes().hash(state);
+    }
+}
+
+impl fmt::Debug for Secp256k1PublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Secp256k1PublicKey({self})")
+    }
+}
+
+impl fmt::Display for Secp256k1PublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", hex::encode(self.to_bytes()))
+    }
+}
+
+#[cfg(feature = "arbitrary")]
+impl<'a> arbitrary::Arbitrary<'a> for Secp256k1PublicKey {
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        Self::from_bytes(u.arbitrary()?).map_err(|_| arbitrary::Error::IncorrectFormat)
+    }
+}
+
+impl Secp256k1PublicKey {
+    pub const SIZE: usize = 33;
+
+    pub fn to_bytes(&self) -> [u8; Self::SIZE] {
+        self.0.to_encoded_point(true).as_ref().try_into().unwrap()
+    }
+
+    pub fn from_bytes(bytes: [u8; Self::SIZE]) -> Result<Self, Error> {
+        Ok(Self(VerifyingKey::from_sec1_bytes(&bytes)?))
+    }
+
+    pub fn verify_prehashed(&self, message_hash: [u8; 32], signature: Secp256k1Signature) -> bool {
```

Comment:
I would think we could take a reference to the `message_hash` as well as the `signature`

Reply:
Fixed